### ISSUE TITLE
seeding does not work when I initialize a linear model even if I do not use it in the code

### ISSUE THREAD

**Issue by MichalisLazarou** [pytorch/pytorch#46149]
I am having trouble seeding my experiments. I realized some very weird behaviour when I initialize a linear layer such as:

`classifier = nn.Linear(640, 5)`

even if this layer is not used at all in the experiments. For example by just initializing this layer and not using it I get results of 87% in an experiment while if I comment this line I get results of 85%. I get exactly the same results for every process (i.e. if I initialize the layer I always get 87% while If I comment the line I always get 85%).

I use the following seeding initialization to make the process deterministic but it seems that initializing this classifier really just changes the results for some reason:

`seed = 16
np.random.seed(seed)
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False
torch.manual_seed(seed)
torch.random.manual_seed(seed)
torch.cuda.manual_seed(seed)'

 What is the matter here?

**Comment by jgericardo** [comment]
When you initialize a `nn.Linear` layer, the weights and bias are also initialized by default with `kaiming_uniform_`.

**Comment by glaringlee** [comment]
@MichalisLazarou 
@jgericardo is correct
